### PR TITLE
Bump metamask/gas-fee-controller to v15.1.1

### PR DESCRIPTION
**Pull request by danjm**
## **Description**

This PR bumps the gas-fee-controller to the latest version.

The changes between the previous version and the latest are:

### [15.1.1]

*Changed**

- Bump `@metamask/polling-controller` to `^6.0.2` ([#4234](https://github.com/MetaMask/core/pull/4234))

**Removed**

- Remove fee history fallback in favour of `eth_gasPrice` call ([#4210](https://github.com/MetaMask/core/pull/4210))

### [15.1.0]

**Added**

- Add nonRPCGasFeeApisDisabled property to the gas fee controller, allowing the user to specify that they want to prevent network request to gas estimate services, and only want gas estimates to be based on rpc requests (eth_feeHistory and eth_gasPrice) to the provider. ([#4094](https://github.com/MetaMask/core.git/pull/4094))

**Fixed**

- Fix GasFeeController incorrectly setting globally selected state, so that state is only updated if the gasFeeEstimate fetched is for the currently selected network ([#4214](https://github.com/MetaMask/core.git/pull/4214))

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/24333?quickstart=1)

## **Related issues**

Part of: https://github.com/MetaMask/MetaMask-planning/issues/2070

## **Manual testing steps**

This is mostly not testable right now. I will add testing steps after some functionality has been added to another PR. Same goes for automated tests. Some things that can be tested:
- Gas related functionality should continue to work unchanged
- Go to the confirmation screen for any transaction on any EIP-1559 network. If developer tools in the background are opened to the network tab, and calls to the [/suggestedGasFees](https://gas.api.cx.metamask.io/networks/1/suggestedGasFees) api are disabled, then you should see requests to a json rpc api with the method `eth_gasPrice`

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by jiexi — DISMISSED** [review]

**Comment by legobeat** [comment]
@metamaskbot update-policies

**Review by vthomas13 — DISMISSED** [review]

File: yarn.lock [hunk 1]
```diff
@@ -5874,6 +5895,23 @@ __metadata:
   languageName: node
   linkType: hard
 
+"@metamask/polling-controller@npm:^6.0.2":
```
[the remark this answers is not in the document]
Comment (legobeat) [reply yarn.lock:5898]: Also, this needs `yarn deduplicate` to dedupe with existing `@metamask/polling-controller@6.0.1`.

https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/78538/workflows/68a63caf-d89e-4bf1-b1c0-ef1ad9665cc9/jobs/2738624

**Review by legobeat — CHANGES_REQUESTED** [review]
Needs a `yarn deduplicate`.

https://github.com/MetaMask/metamask-extension/pull/24333/files#r1586616380

**Comment by legobeat** [comment]
I guess this is waiting for https://github.com/MetaMask/core/pull/4245 and a release of v15.1.2?

**Comment by danjm** [comment]
this was replaced by https://github.com/MetaMask/metamask-extension/pull/24520

**Pull request closed without merging**